Browser_IDE/executionEnvironment.js

```diff
@@ -312,4 +312,94 @@ class ExecutionEnvironment extends EventTarget{
         return iframe;
     }
 
-}
\ No newline at end of file
+}
+
+document.getElementById("stopProgram").addEventListener("click", () => {
+    terminateProgram();
+});
+
+document.getElementById("collapsedStopProgram").addEventListener("click", () => {
+    terminateProgram();
+});
+
+// Attach event listeners for Stop buttons
+initializeEventListeners();
+
+function initializeEventListeners() {
+    const stopButton = document.getElementById("stopProgram");
+    const collapsedStopButton = document.getElementById("collapsedStopProgram");
+
+    stopButton.addEventListener("click", handleStopProgram);
+    collapsedStopButton.addEventListener("click", handleStopProgram);
+}
+
+/**
+ * Handles the stop button click event
+ */
+function handleStopProgram() {
+    if (!isProgramRunning()) {
+        showNotification("No program is currently running.", "warning");
+        return;
+    }
+
+    try {
+        terminateProgram();
+        showNotification("Program stopped successfully.", "info");
+    } catch (error) {
+        console.error("Error while stopping the program:", error);
+        showNotification("Failed to stop the program. Please try again.", "error");
+    }
+}
+
+/**
+ * Terminates the currently running program
+ */
+function terminateProgram() {
+    try {
+        // Logic to terminate the running process
+        runtimeManager.terminate();
+        updateRuntimeButtons(false); // Disable runtime buttons after stopping
+    } catch (error) {
+        throw new Error("Program termination failed: " + error.message);
+    }
+}
+
+/**
+ * Updates the runtime control buttons based on the program state
+ * @param {boolean} isRunning - Whether the program is currently running
+ */
+function updateRuntimeButtons(isRunning) {
+    toggleButtonState("stopProgram", isRunning);
+    toggleButtonState("collapsedStopProgram", isRunning);
+}
```

Comment:
This should all be in `editorMain.js` - there's a lot of framework code already there to help. Have a look in the functions `setupIDEButtonEvents` and `updateButtons` :smiley: